### PR TITLE
Add Tenancy to Rack Reservations; Fixes #1592

### LINKED ISSUE #1592: Add Tenant ownership to Rack Reservations
### Issue type

[x] Feature request 
[ ] Bug report      
[ ] Documentation   

### Description
I propose an addition to the rack reservation table that allows association with a Tenant (Exactly like IP addresses) which would allow racks used for "Shared Colocation Environments" to have the ability to reserve less than an entire rack for a customer. E.g. 7-8U per customer. 

This would not matter if they were managed as devices are able to be assigned. But for environments where you only are keeping track of what section you have assigned the Tenant, it would be very helpful.

### PR DESCRIPTION
**Pull request by dirtycajunrice**
### Fixes:
#1592 

Added tenant ownership, and created rack reservation stat on tenancy page, appended to api

### CODE REVIEW

**Comment by DirtyCajunRice** [comment]
Resolved the 2 conflicts

**Comment by DirtyCajunRice** [comment]
@jeremystretch Ok boss this is ready to go/for review!

**Comment by DirtyCajunRice** [comment]
What is the normal timeline for merge requests?

File: netbox/dcim/migrations/0051_merge_20171031_2024.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
```
Comment (jeremystretch) [inline netbox/dcim/migrations/0051_merge_20171031_2024.py]: Where did this migration come from? It's empty and shoudn't be needed.

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Looks like you have an extraneous migration, but other than that this looks good.

File: netbox/dcim/migrations/0051_merge_20171031_2024.py (hunk 1 above)
Comment (DirtyCajunRice) [reply netbox/dcim/migrations/0051_merge_20171031_2024.py]: This Migration was a requirement for the 49 migration you created at the same time that i created one. It merges your 49 and my 50. Is there a better way to do this?

**Comment by DirtyCajunRice** [comment]
Was getting:
CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0050_rackreservation_tenant, 0049_rackreservation_change_user in dcim).
To fix them run 'python manage.py makemigrations --merge'
And merged originally to fix. But realized that just changing the dependency to be linear with yours fixes it too.

**Review by jeremystretch — APPROVED** [review]

**Pull request merged**